bode/bode/models/task_relations_actions.py

```diff
@@ -0,0 +1,57 @@
+from bode.app import db
+from bode.models.task_relation import RelationType, TaskRelation
+from bode.resources.task_relations.schemas import DirectedRelationType
+
+
+def get_lhs_related_tasks(task_id, filters=list()):
+    from bode.models.task import Task
+
+    all_filters = [TaskRelation.first_task_id == task_id] + filters
+    return (
+        db.session.query(TaskRelation, Task)
+        .filter(*all_filters)
+        .join(Task, TaskRelation.second_task_id == Task.id)
+        .all()
+    )
+
+
+def get_rhs_related_tasks(task_id, filters=list()):
+    from bode.models.task import Task
+
+    all_filters = [TaskRelation.second_task_id == task_id] + filters
+    return (
+        db.session.query(TaskRelation, Task)
+        .filter(*all_filters)
+        .join(Task, TaskRelation.first_task_id == Task.id)
+        .all()
+    )
+
+
+def get_related_tasks(task_id):
+    return get_lhs_related_tasks(task_id) + get_rhs_related_tasks(task_id)
+
+
+def map_to_related_task_schema(relation: TaskRelation, task_id):
+    match relation.type:
+        case RelationType.Dependent.value:
+            type = (
+                DirectedRelationType.Blocks.value
+                if task_id == relation.first_task_id
+                else DirectedRelationType.IsBlockedBy.value
+            )
+        case RelationType.Subtask.value:
+            type = (
+                DirectedRelationType.Supertask.value
+                if task_id == relation.first_task_id
+                else DirectedRelationType.Subtask.value
+            )
+        case _:
+            type = DirectedRelationType.Interchangable.value
+
+    return type
+
+
+def get_relation_types(task_id):
+    filters = [TaskRelation.first_task_id == task_id or TaskRelation.second_task_id == task_id]
+    relations = TaskRelation.query.filter(*filters).all()
```

Comment:
This query isn't working. I suggest use `get_related_tasks` function which uses `get_rhs` and `get_lhs` or something like that.

Reply:
EXAMPLE:
I created task "Supertask of subtask" and add to this task one subtask "subtask".
So in database are one relation and two tasks.
![image](https://user-images.githubusercontent.com/72767517/167830444-d483259d-e95b-439a-9501-31b7e6dfe6b6.png)
![image](https://user-images.githubusercontent.com/72767517/167830645-9898016a-11d5-45aa-9a87-03ef1c0be4d8.png)

In the second picture we can see tasks, which I created, but supertask haven't got flag 'supertask'. The same situation there is with blocking relation. Only we can get 'is_blocking_by' flag.